### PR TITLE
Added script to trigger based on TH filters.

### PR DESCRIPTION
**Pull request by adusca**
This script is based on @jgraham use case.

This is an example output:
https://pastebin.mozilla.org/8832772

I added --limit with default 100 to protect against accidentally triggering to many things (e.g. forgetting to put a repo name in --includes)

### CODE REVIEW

File: scripts/th_filters.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    """Parse command line options."""
+    parser = ArgumentParser()
+
+    # Required arguments
+
+    parser.add_argument("repo",
+                        help='Branch name')
+
+    parser.add_argument("rev",
+                        help='The 12 character represneting a revision (most recent).')
```
Comment (armenzg) [inline scripts/th_filters.py]: Typo in representing.

File: scripts/th_filters.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    buildernames = fetch_allthethings_data()['builders'].keys()
+
+    for word in filters:
+        buildernames = filter(lambda x: word in x, buildernames)
+
+    if len(buildernames) > options.limit:
+        LOG.info('There %i matching buildernames, the limit is %i. If you really want'
+                 'to trigger everything, try again with --limit %i.' % (len(buildernames),
+                                                                        options.limit, options.limit))
+        exit(1)
```
Comment (armenzg) [inline scripts/th_filters.py:97]: Extra white line.

File: scripts/th_filters.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    for buildername in buildernames:
+        trigger_range(
+            buildername=buildername,
+            revisions=[options.rev],
```
Comment (armenzg) [inline scripts/th_filters.py:102]: We're not really triggering a range, however, we can add that support later if need be.

**Comment by armenzg** [comment]
My apologies on the delay of this review.
This looks good to me.

Could you also add the support for --exclude? (being optional)

**Comment by armenzg** [comment]
@adusca is this ready for review? I see one more commit since my last comment.

**Comment by adusca** [comment]
Yes! Sorry, I forgot to leave a comment yesterday

File: scripts/th_filters.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        LOG.setLevel(logging.DEBUG)
+        logging.getLogger("requests").setLevel(logging.DEBUG)
+        LOG.info("Setting DEBUG level")
+    else:
+        LOG.setLevel(logging.INFO)
+        # requests is too noisy and adds no value
+        logging.getLogger("requests").setLevel(logging.WARNING)
+
+    filters_in = options.includes.split(' ') + [options.repo]
+    buildernames = fetch_allthethings_data()['builders'].keys()
```
Comment (armenzg) [inline scripts/th_filters.py]: Could you please call query_builders() from mozci which is a wrapper to that?

**Comment by armenzg** [comment]
After that, feel free to merge it. Thanks Alice!

**Pull request merged**